### ISSUE TITLE
`no-regex-spaces`: Add RegExp object form

### ISSUE THREAD

**Issue by IanVS** [eslint/eslint#3586]
The docs for this rule specifically call out that the following will not be considered a warning:

```js
/*eslint no-regex-spaces: 2*/

var re = new RegExp("foo   bar");
```

This seems to be a limitation, as it is not asserted in the tests, and intuitively it seems that the RegExp object format should be treated the same way as a literal, multiple spaces are equally hard to read in both.

**Comment by nzakas** [comment]
This rule is only designed to test regex literals. It's really hard to do anything for the constructor since you can pass non-literals or expressions.

**Comment by IanVS** [comment]
Wouldn't it at least be possible to check for spaces in literals within the constructor?  Failing that, I think the docs could be clarified that it is a known limitation of the rule, rather than by-design.

**Comment by nzakas** [comment]
I'm not against adding it, just pointing out that this rule only checked literals by design. It's not a bug. It was intended to map to what JSLint and JSHint do, which only check literals. We can also check strings.

> Failing that, I think the docs could be clarified that it is a known limitation of the rule, rather than by-design.

I have no idea what this means. :)

**Comment by IanVS** [comment]
From the [rule docs](http://eslint.org/docs/rules/no-regex-spaces):

> The following patterns are considered warnings:
> 
> ```js
> var re = /foo   bar/;
> ```
> 
> The following patterns are not warnings:
> 
> ```js
> var re = new RegExp("foo   bar");
> ```

In most of the docs, the "not warnings" section is used to show how to "fix" the code so that the rule will no longer flag.  So, what I'm proposing is:

> The following patterns are considered warnings:
> 
> ```js
> var re = /foo   bar/;
> ```
> 
> The following patterns are not warnings:
> 
> ```js
> var re = /foo {3}bar/;
> ```
>
> ### Known Limitations
>
> This rule does not apply to the constructor form of `RegExp`, and therefore the following will not warn:
> 
> ```js
> var re = new RegExp("foo   bar");
> ```

**Comment by nzakas** [comment]
"Not warnings" are definitely not "fixes", they are designed to show patterns that won't be flagged either because they are correct or they are not covered by the rule. They are provided to avoid confusion of people saying, "should this be flagged?"

**Comment by alberto** [comment]
@IanVS are you still interested in implementing this?

**Comment by IanVS** [comment]
Yeah, this dropped off my radar.  I'll submit a docs PR within the next day or two.

**Comment by jacksonrayhamilton** [comment]
@IanVS, I'd also like to have this functionality. If you'd like, I could implement it too.

**Comment by IanVS** [comment]
@jacksonrayhamilton Sorry, I guess I'm not exactly sure what enhancement you're proposing.  Are you saying you want to teach `no-regex-spaces` to flag spaces in literals within `RexExp` constructors?

**Comment by jacksonrayhamilton** [comment]
Yep!

**Comment by IanVS** [comment]
Cool, I'd be all for that, and @nzakas has said

> I'm not against adding it

So I'd say go for it. :)
